### PR TITLE
(PIE-478) Acceptance Test for Token Auth

### PR DESCRIPTION
**Pull request by RandomNoun7**
This change adds an acceptance test for Token Authorization module for
PE. As part of this change the tests were made a little more flexible
and composable to make it easier to add future tests and to make this
new test work.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -51,7 +51,7 @@
   }
 
   if $include_api_collection {
-    if ($pe_username == undef) or ($pe_password == undef) or ($pe_console == undef) {
+    if ($pe_username == undef) or ($pe_console == undef) {
```
Comment (gsparks) [inline manifests/init.pp:54]: Is the username required if they are providing a token? Should it be something like if (($pe_username == undef) and ($pe_token == undef)) or ($pe_console == undef)?

File: spec/spec_helper_acceptance_local.rb [hunk 2]
```diff
@@ -58,22 +79,97 @@ def return_host
...
   MANIFEST
 end
+
+def token_splunk_class
+  <<-MANIFEST
+    class{'splunk_hec':
+      url                    => "http://#{master.uri}:8088/services/collector/event",
+      pe_console             => "#{master.uri}",
+      splunk_token           => 'abcd1234',
+      pe_username            => 'admin',
```
Comment (gsparks) [inline spec/spec_helper_acceptance_local.rb:103]: is pe_username required here?

**Review by gsparks — CHANGES_REQUESTED** [review]
Looks good other than I don't think we need to user if the token is provided.

File: manifests/init.pp (hunk 1 above)
Comment (RandomNoun7) [reply manifests/init.pp]: Good point. Rejiggered it to get rid of the `pe_console` check that we don't need to do anymore either.

File: spec/spec_helper_acceptance_local.rb (hunk 2 above)
Comment (RandomNoun7) [reply spec/spec_helper_acceptance_local.rb]: Nope! And removing it helps make sure we catch edge cases related to that too.

**Review by gsparks — APPROVED** [review]

**Pull request merged**